substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationAWTSupport.java

```diff
@@ -79,7 +83,10 @@ public void beforeImageWrite(BeforeImageWriteAccess access) {
             jniRegistrationSupport.addJavaShimExports(
                             "JNU_ThrowClassNotFoundException");
         }
-        if (jniRegistrationSupport.isRegisteredLibrary("fontmanager") && JavaVersionUtil.JAVA_SPEC >= 11) {
+        // harfbuzz became a separate library in OpenJDK 16/11.0.10 (JDK-8249821) and then went back
+        // to be included in fontmanager library in OpenJDK 17/11.0.13 (JDK-8255790)
+        if (jniRegistrationSupport.isRegisteredLibrary("fontmanager") &&
+                        ((JavaVersionUtil.JAVA_SPEC == 11 && JDK_UPDATE >= 10 && JDK_UPDATE < 12) || JavaVersionUtil.JAVA_SPEC == 16)) {
```

Comment:
Should this be `JDK_UPDATE <= 12` to match the JDK range in which harfbuzz was a separate library?

Reply:
Ah, yes, nice catch!

Reply:
Fixed.